WordPress/Classes/ViewRelated/Comments/CommentContentTableViewCell.swift

```diff
@@ -0,0 +1,221 @@
+import UIKit
+
+class CommentContentTableViewCell: UITableViewCell, NibReusable {
+
+    // determines the state of the like button.
+    enum LikeState {
+        case unliked
+        case liked
+    }
+
+    // all the available images to display for the accessory button.
+    enum AccessoryButtonType {
+        case share
+        case threeDots
+    }
+
+    // MARK: - Public Properties
+
+    var displayName: String? = "" {
+        didSet {
+            nameLabel?.setText(displayName ?? "")
+        }
+    }
+
+    var displayDate: String? = "" {
+        didSet {
+            dateLabel?.setText(displayDate ?? "")
+        }
+    }
+
+    var accessoryButtonType: AccessoryButtonType = .share {
+        didSet {
+            accessoryButton.setImage(accessoryButtonImage, for: .normal)
+        }
+    }
+
+    var numberOfLikes: Int = 0 {
+        didSet {
+            updateLikeButton()
+        }
+    }
+
+    var likeButtonState: LikeState = .unliked {
+        didSet {
+            updateLikeButton()
+        }
+    }
+
+    // MARK: Button Tap Handlers
+
+    var nameLabelTapAction: (() -> Void)? = nil
```

Comment:
Maybe I missed it, but I don't see that tapping the user's name should do anything. Is this needed/what will this do?

Reply:
The current (old) comment detail shows the author's site in a modal window. This is added to support that capability.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17109/commits/d73aec748113eee66b945c205d05b8944e29b664!